bin/rucio-admin

```diff
@@ -839,11 +841,35 @@ def declare_bad_file_replicas(args):
                     bad_files.append(bad_file)
     else:
         bad_files = args.listbadfiles
-    for chunk in chunks(bad_files, 500):
-        non_declared = client.declare_bad_file_replicas(pfns=chunk, reason=args.reason)
-        for rse in non_declared:
-            for pfn in non_declared[rse]:
-                print '%s : PFN %s cannot be declared.' % (rse, pfn)
+
+    # Interpret filenames not in scheme://* format as LFNs and convert them to PFNs
+    bad_files_pfns = []
+    for bad_file in bad_files:
+        if bad_file.find('://') == -1:
+            scope, name = extract_scope(bad_file)
+            did_info = client.get_did(scope, name)
+            if did_info['type'].upper() != 'FILE' and not args.allow_collection:
+                print 'DID %s:%s is a collection and --allow-collection was not specified.' % (scope, name)
+                return FAILURE
+            replicas = [replica for rep in client.list_replicas([{'scope': scope, 'name': name}])
+                        for replica in rep['pfns'].keys()]
+            bad_files_pfns.extend(replicas)
+        else:
+            bad_files_pfns.append(bad_file)
+    print "PFNs that will be declared bad:"
+    for pfn in bad_files_pfns:
+        print pfn
+
```

Comment:
Maybe you should remove the print statement there, or at least make them optional

Reply:
Ok, suppressed when not using --verbose